controllers/repository_controller.go

```diff
@@ -291,3 +313,22 @@ func (r RepositoryReconciler) ensureRatingServiceAccount(ctx context.Context, na
 
 	return nil
 }
+func (r *RepositoryReconciler) getRepoType(apiURL, auth string) (*http.Response, error) {
```

Comment:
As you named this func `getRepoType`, you should return the repository types rather than a http.Response.